homeassistant/components/opengarage/config_flow.py

```diff
@@ -0,0 +1,81 @@
+"""Config flow for OpenGarage integration."""
+import logging
+
+import aiohttp
+import opengarage
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_VERIFY_SSL
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_DEVICE_KEY, DEFAULT_PORT, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE_KEY): str,
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_VERIFY_SSL, default=False): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    open_garage = opengarage.OpenGarage(
+        f"{data[CONF_HOST]}:{data[CONF_PORT]}",
+        data[CONF_DEVICE_KEY],
+        data[CONF_VERIFY_SSL],
+        async_get_clientsession(hass),
+    )
+
+    status = await open_garage.update_state()
+    if status is None:
+        raise InvalidAuth
+
+    return {"title": status.get("name"), "unique_id": format_mac(status["mac"])}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for OpenGarage."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            info = None
+            try:
+                info = await validate_input(self.hass, user_input)
+            except (aiohttp.ClientError, aiohttp.ClientConnectorError):
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if info:
+                await self.async_set_unique_id(info["unique_id"])
+                self._abort_if_unique_id_configured(
+                    updates={CONF_HOST: user_input[CONF_HOST]}
```

Comment:
Why do we update host in this case? This is not a pattern we use in user flows just in discovery flows.